cmake/list_insert_options.cmake

```diff
@@ -0,0 +1,55 @@
+#
+# Append key[=value] elements from one list to another list if they are not already in the target list.
```

Comment:
Well, I guess that is the problem of trying to set different values (coming from different packages). You already have the same problem with two different values both coming from dependencies because you have little control in which order they are being found.

I think it is just counter intuitive if this function uses the first value.